### PR TITLE
Allow PRs from forks to run tests

### PR DESCRIPTION
**Pull request by Akshat2Jain**
### Description

This pr aims to allow tests to run on PRs from forks

### How

- putting `pull_request` in on in all `.yml` files here:
- ` if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name != github.event.pull_request.base.repo.full_name` in all jobs.

This pr is reated to isse #1656

### CODE REVIEW

File: .github/workflows/black.yml [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
+on: [push,pull_request]
 jobs:
   build:
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
         python-version: [3.8]
+        
+    if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name != github.event.pull_request.base.repo.full_name
```
Comment (wesleybl) [inline .github/workflows/black.yml:11]: I think it's better for the test to be the first thing in the job.

File: .github/workflows/update_remote_docs.yml [hunk 2]
```diff
@@ -6,14 +6,21 @@ on:
       - "master"
     paths:
       - "docs/*"
+  pull_request:
+    branches:
+      - "master"
+    paths:
+      - "docs/*"
+
```
Comment (wesleybl) [inline .github/workflows/update_remote_docs.yml:14]: These lines can be removed. The update must be done only in the push on the master.

**Review by wesleybl — CHANGES_REQUESTED** [review]

**Review by wesleybl — CHANGES_REQUESTED** [review]
Please add an internal entry in changes.

**Comment by Akshat2Jain** [comment]
Hey @wesleybl, I have made the changes. Can you review it.

File: news/1659.internal [hunk 3]
```diff
@@ -0,0 +1 @@
+allow tests to run on PRs from forks @akshat2jain
```
Comment (wesleybl) [inline news/1659.internal:1]: Use the issue id (1656) in the file name.

```suggestion
Allow GHA tests to run on PRs from forks. @Akshat2Jain
```

**Review by wesleybl — CHANGES_REQUESTED** [review]

File: .github/workflows/update_remote_docs.yml [hunk 4]
```diff
@@ -6,14 +6,16 @@ on:
       - "master"
     paths:
       - "docs/*"
+  
 
 jobs:
   deploy:
+    if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name != github.event.pull_request.base.repo.full_name
```
Comment (wesleybl) [inline .github/workflows/update_remote_docs.yml:13]: Here this line can be removed, as this job does not run in PR.

File: .github/workflows/black.yml [hunk 5]
```diff
@@ -1,13 +1,14 @@
...
 jobs:
   build:
+    if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name != github.event.pull_request.base.repo.full_name
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
         python-version: [3.8]
-
+
```
Comment (wesleybl) [inline .github/workflows/black.yml:11]: Please, do not add spaces on the lines.

**Comment by Akshat2Jain** [comment]
Now I think, Its good to go @wesleybl .

File: .github/workflows/black.yml [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
+on: [push,pull_request]
 jobs:
   build:
+    if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name != github.event.pull_request.base.repo.full_name
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
         python-version: [3.8]
-
```
Comment (wesleybl) [inline .github/workflows/black.yml:10]: Please do not remove line spacing. The spacing must be keep as it was before.

**Review by wesleybl — CHANGES_REQUESTED** [review]

**Comment by Akshat2Jain** [comment]
@wesleybl Can you review it again? Thanks.

**Review by wesleybl — APPROVED** [review]
LGTM

**Comment by Akshat2Jain** [comment]
> @Akshat2Jain by having several commits, I suspect that you were making the commits through the github website. I recommend that next time, you make a clone of the repository and put several files in one commit. That way we won't have multiple commits.

Yeah, I will keep that in mind!
Thank You

**Review by davisagli — APPROVED** [review]
@Akshat2Jain @wesleybl Thanks, this looks good!

**Pull request merged**